### DISCUSSION TITLE
2️⃣ 0️⃣ 2️⃣ 4️⃣  New Year, New Commitments, and New Actions 🏁

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#84197] [Actions]
![Screenshot 2024-01-03 at 5 49 46 PM](https://github.com/community/community/assets/95834130/373d0423-9a9e-4a51-87ad-7c877c17472d)

> Alt Text: Octocat wearing a spacesuit surrounded by a globe and stars.

Hello everyone! 👋 and welcome to the first Action-packed check-in of 2️⃣ 0️⃣ 2️⃣ 4️⃣ (2024). Our feature list is getting stronger each day and you don’t wanna miss out on what I’ve for you today.  

As always with all our check-ins, we can’t get the new year started without thanking our community champions enough. In 2023, the Actions community had a number of new faces contributing regularly. Let’s cheer some of the community members for their valuable inputs in December. 

- [@TaCqz](https://github.com/TaCqz) Was looking to run an action whenever a specific branch is created, [@GETrackerDan](https://github.com/GETrackerDan) helped [narrowing down to the root cause](https://github.com/orgs/community/discussions/80704) that helped [@TaCqz](https://github.com/TaCqz) solve the problem. Kudos to their amazing collaboration 👏
- Great job ✨  by @ChristopherHX for being a top contributor in the Actions space. Though they have contributed to multiple discussions, the one that really stands out in December where they helped another user get past [download artifact workflow](https://github.com/orgs/community/discussions/81545). 
- December 2023 top Self-Solvers 🟢 
    - [@BearToCode](https://github.com/BearToCode) couldn’t run an action workflow that used prettier. However, they soon diagnosed that [prettier needs an update to v3.1.1](https://github.com/orgs/community/discussions/79380) which helped them solve the problem. Thanks for sharing it with the wider community. ❤️ 
    - Let’s look at this self-solve ✔️ - **Much deserving** 👏 applause for [@lorennorman](https://github.com/lorennorman) for updating us with the thorough response and calling out the need to update the [documentation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path) about [$GITHUB_PATH not working on macOS or Windows runners](https://github.com/orgs/community/discussions/80916). 

Remember we love open source, you can always [contribute](https://github.com/github/docs/edit/main/content/index.md) directly to any doc by submitting a PR, learn more about [how to contribute](https://github.com/github/docs/blob/main/CONTRIBUTING.md) to docs. 📝 

#### [Artifacts v4 is now Generally Available](https://github.blog/changelog/2023-12-14-github-actions-artifacts-v4-is-now-generally-available/) 👀 

We listened to [your feedback](https://github.com/actions/upload-artifact/issues/199) and released new versions (`v4`) of [actions/upload-artifact](https://github.com/actions/upload-artifact) and [actions/download-artifact](https://github.com/actions/download-artifact). While this version of the artifact actions includes up to 10x performance improvements and several new features, **there are also [key differences](https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes) from previous versions that may require updates to your workflows**.

Note that Artifacts v4 is only available to GitHub.com customers today but we will be extending support to GitHub Enterprise Server (GHES) customers in the future.

### Top resources to power up 🍄 your learning journey for 2024

-  **GitHub’s top blog posts of 2023**: With the books 📚  closed on 2023, we're highlighting some of the incredible work from GitHub’s engineers, product teams, and security researchers in [this article](https://github.blog/2023-12-27-githubs-top-blog-posts-of-2023/).
- **Interested in exploring DecSecOps** for your enterprise? Unwrap the enterprise guide to [AI-Powered DevSecOps](https://resources.github.com/enterprise-guide-ai-powered-devsecops/#form) to learn how to help your organization integrate security measures throughout your software supply chain.
- **Level up 🆙  your GitHub Actions skillset** with our learning pathways on automation. Unlock 🔓 the full potential of GitHub with our specialized **learning pathways**. Each pathway is designed to help you master a new topic—automation, security, and administration & governance. With clear, actionable insights 📊  , you’ll refine your skills and elevate your projects. Whether enhancing development workflows, securing applications, or streamlining governance processes, these pathways are tailored to ensure you meet your needs efficiently and effectively. [Get started with your learning pathway today](https://resources.github.com/learn/pathways/?utm_source=Discussion&utm_medium=GitHub) ⏩ 

What are some of the Actions you’re going to take to make 2024 a year to remember?

**Comment by thisara011** [comment]
good explain

**Comment by jsoref** [comment]
Well...
1. it'd be nice if someone actually maintained the [actions/runner](https://github.com/actions/runner/pulls) repository
   - speaking from [personal experience](https://github.com/actions/runner/pulls/jsoref) it is not maintained
2. it'd be nice if someone actually maintained the [actions/toolkit](https://github.com/actions/toolkit/pulls) repository
   - speaking from [personal experience](https://github.com/actions/runner/pulls/jsoref) it is not maintained
   - https://github.com/actions/toolkit/pull/1571 is especially frustrating
3. it'd be nice if someone actually maintained the [actions/starter-workflows](https://github.com/actions/starter-workflows/pulls) repository
   - https://github.com/actions/starter-workflows/pull/2270 (for example)

I'd love to get [check-spelling](https://github.com/marketplace/actions/check-spelling) added to [starter-workflows](https://github.com/actions/starter-workflows), but I can't tell what the rules are for admission, and the last N times I looked at the docs, the partner program requirements were unreasonable for individuals. I'm not a company, so I guess I can't?

I'd really love to speak with a human about various pain points in the actions ecosystem.

This doesn't even cover the fact that user repositories can't track active workflows -- which means that no-one in their right mind would enable actions in their own personal forks (because if they do, they won't know which ones are running). I'm supposed to be looking for an open discussion on this pain point, but I don't have time at this time. So, maybe a human can talk to me and we can discuss various pain points. I'm quite reachable.

**Comment by nkhandwe** [comment]
Glade to see this
